internal/core/metadata/rest_deviceprofile.go

```diff
@@ -150,28 +150,25 @@ func restGetProfileByProfileId(w http.ResponseWriter, r *http.Request) {
 
 func restDeleteProfileByProfileId(w http.ResponseWriter, r *http.Request) {
 	vars := mux.Vars(r)
-	var did string = vars["id"]
+	var did = vars["id"]
 
-	// Check if the device profile exists
-	dp, err := dbClient.GetDeviceProfileById(did)
+	op := device_profile.NewDeleteByIDExecutor(dbClient, did)
+	err := op.Execute()
 	if err != nil {
-		if err == db.ErrNotFound {
+		LoggingClient.Error(err.Error())
+		switch err.(type) {
+		case errors2.ErrDeviceProfileNotFound:
```

Comment:
Something to consider -- It looks to me like we have custom error types in `internal/core/metadata/errors/types.go` that we could use in this file instead of `errors.New("blah")` If that's the case, then the need for an `errors2` alias goes away.

For example, line 42
`err = errors.New("Max limit exceeded with request for profiles")`
We have an `ErrLimitExceeded` type we could use instead.

Your call as to whether to include such an audit (for this file only) in this PR. If not, please create an issue to follow-up.

Reply:
I'd like to tackle this now while I am in here. Less overhead rather than creating an issue, new PR, etc.